### ISSUE TITLE
VS fails to extract info from CMake

### ISSUE THREAD

**Issue by dvirtz** [Microsoft/VSLinux#190]
* A brief description
I'm trying to build a CMake based project on WSL. The project builds successfully but the build hangs after `Starting CMake target info extraction`.
After pressing `Cancel` I get the following error:
```C#
Object reference not set to an instance of an object.
   at libcmake.Linux.RemoteCMakeDaemonDriver.StopDaemon()
   at libcmake.CMakeDaemonDriver.Dispose(Boolean disposing)
   at libcmake.CMakeDaemonDriver.Dispose()
   at libcmake.Linux.RemoteCMakeLists.Dispose()
   at libcmake.Linux.LocalCMakeLists.Dispose()
   at Microsoft.VisualStudio.CMake.Project.CMakeListsUtils.<ParseAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Microsoft.VisualStudio.CMake.Project.CMakeListsCache.<ParseAsync>d__28.MoveNext()
```
* Versions
VC++ 15.4.0 Preview 3.0
GCC 6.2
WSL with trusty
CMake 3.9.0

* Steps required to reproduce the error
Build https://github.com/dvirtz/GreenTiger on Linux-Debug or Linux-Release configurations

**Comment by lukka** [comment]
@dvirtz while trying to reproduce your scenario with the next to be released preview, I got the following error from hunter. Do you know how to get past that error?


```
1> 4:44:17 PM: Finished copying files (elapsed time 00h:00m:22s:340ms).
1> /usr/local/bin/cmake -G "Unix Makefiles"  -DCMAKE_BUILD_TYPE="Debug" "\var\tmp\src\14c646ea-54c8-6f3c-848d-e5896a3d3fc3\Linux-Debug"
1> -- [hunter] Initializing Hunter workspace (8898dd07756c408c63c63add903c8600c4cf4c0e)
1> -- [hunter]   https://github.com/ruslo/hunter/archive/v0.19.77.tar.gz
1> -- [hunter]   -> /home/xxx/.hunter/_Base/Download/Hunter/0.19.77/8898dd0
1> CMake Error at Build/Hunter-prefix/src/Hunter-stamp/download-Hunter.cmake:157 (message):
1>   Each download failed!
1> 
1>     error: downloading 'https://github.com/ruslo/hunter/archive/v0.19.77.tar.gz' failed
1>          status_code: 1
1>          status_string: "Unsupported protocol"
1>          log:
1>          --- LOG BEGIN ---
1>          Protocol "https" not supported or disabled in libcurl
1> 
1>   Closing connection -1
```

**Comment by dvirtz** [comment]
Yes, you need to build cmake with system curl.

**Comment by lukka** [comment]
The problem has been fixed, with the next VS version 15.5 (ETA non available yet) I could successfully build&debug the chapter 1 of GreenTiger repository. Closing issue.


```
shellexec 1.0
Copyright (C) 2016 Microsoft Corporation. All rights reserved.
Welcome to Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-43-Microsoft x86_64)
 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage
  Get cloud support with Ubuntu Advantage Cloud Guest:
    http://www.ubuntu.com/business/services/cloud
 packages can be updated.
 updates are security updates.
Last login: Tue Oct  3 11:34:32 2017 from ::1
luca@luca-desktop:~$ =thread-group-added,id="i1"
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/gdb...(no debugging symbols found)...done.
=cmd-param-changed,param="pagination",value="off"

Breakpoint 1, main (argc=1, argv=0x7ffffffdeba8) at /home/luca/.hunter/_Base/8898dd0/e1266bb/e19651d/Install/include/catch.hpp:11157
11157	int main (int argc, char * argv[]) {
8
7
80
===============================================================================
All tests passed (15 assertions in 4 test cases)
[Inferior 1 (process 12731) exited normally]
The thread 'Chapter01' (0x31bb) has exited with code 0 (0x0).
The program 'Chapter01.exe' has exited with code 0 (0x0).

```
